lib/src/dart3_suggestors/null_safety_prep/connect_required_props.dart

```diff
@@ -0,0 +1,144 @@
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:collection/collection.dart';
+import 'package:over_react_codemod/src/dart3_suggestors/null_safety_prep/utils/props_utils.dart';
+import 'package:over_react_codemod/src/util.dart';
+import 'package:over_react_codemod/src/util/class_suggestor.dart';
+
+import 'analyzer_plugin_utils.dart';
+
+/// Suggestor that adds `@Props(disableRequiredPropValidation: {...})` annotations
+/// for props that are set in `connect` components.
+class ConnectRequiredProps extends RecursiveAstVisitor with ClassSuggestor {
+  /// Running list of props that should be ignored per mixin that will all be added
+  /// at the end in [generatePatches].
+  final _ignoredPropsByMixin = <InterfaceElement, List<String>>{};
+
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    super.visitCascadeExpression(node);
+
+    // Verify the builder usage is within the `connect` method call.
+    final connect = node.thisOrAncestorMatching<MethodInvocation>(
+        (n) => n is MethodInvocation && n.methodName.name == 'connect');
+    if (connect == null) return;
+
+    // Verify the builder usage is within one of the targeted connect args.
+    final connectArgs =
+        connect.argumentList.arguments.whereType<NamedExpression>();
+    final connectArg = node.thisOrAncestorMatching<NamedExpression>((n) =>
+        n is NamedExpression &&
+        connectArgs.contains(n) &&
+        connectArgNames.contains(n.name.label.name));
+    if (connectArg == null) return;
+
+    final cascadedProps = getCascadedProps(node).toList();
+
+    for (final field in cascadedProps) {
+      final propsElement =
+          node.staticType?.typeOrBound.tryCast<InterfaceType>()?.element;
+      if (propsElement == null) continue;
+
+      // Keep a running list of props to ignore per props mixin.
+      final fieldName = field.name.name;
+      if (_ignoredPropsByMixin[propsElement] != null) {
+        if (!_ignoredPropsByMixin[propsElement]!.contains(fieldName)) {
+          _ignoredPropsByMixin[propsElement]!.add(fieldName);
+        }
+      } else {
+        _ignoredPropsByMixin[propsElement] = [fieldName];
+      }
+    }
+  }
+
+  @override
+  Future<void> generatePatches() async {
+    _ignoredPropsByMixin.clear();
+    final result = await context.getResolvedUnit();
+    if (result == null) {
+      throw Exception(
+          'Could not get resolved result for "${context.relativePath}"');
+    }
+    result.unit.accept(this);
+
+    // Add the patches at the end so that all the props to be ignored can be collected
+    // from the different args in `connect` before adding patches to avoid duplicate patches.
+    for (final propsClass in _ignoredPropsByMixin.keys) {
+      final propsToIgnore = _ignoredPropsByMixin[propsClass]!;
```

Comment:
#nit In general, when iterating over Map key/value pairs, I think it's a good idea to use null-safe APIs, such as `.forEach` or iterating over `.entries`. 

That way, there's no risk of `!` being incorrectly applied to value if it's nullable (e.g., if this map was typed as Map<InterfaceElement, List<String>?>`).